helix-term/src/commands.rs

```diff
@@ -4845,3 +4846,75 @@ fn replay_macro(cx: &mut Context) {
         cx.editor.macro_replaying.pop();
     }));
 }
+
+fn toggle_setting(cx: &mut Context) {
+    cx.on_next_key(move |cx, event| {
+        use helix_view::editor::ConfigEvent;
+
+        cx.editor.autoinfo = None;
+        cx.editor.pseudo_pending = None;
+
+        if let Some(ch) = event.char() {
+            let mut config = cx.editor.config().clone();
+
+            match ch {
+                'w' => {
+                    use helix_view::editor::{
+                        WhitespaceRender::Basic, WhitespaceRenderValue::All,
+                        WhitespaceRenderValue::None,
+                    };
+
+                    config.whitespace.render = match config.whitespace.render {
+                        Basic(None) => Basic(All),
+                        _ => Basic(None),
+                    };
+                }
+                'i' => config.indent_guides.render = !config.indent_guides.render,
+                'c' => config.cursorline = !config.cursorline,
+                'r' => {
+                    config.rulers = match config.rulers.as_slice() {
+                        [80] => vec![],
```

Comment:
If the user has a ruler defined, then the toggle will not hide it but rather move it to 80.